examples/nrf_802154_delayed_echo_client/src/nrf_802154_delayed_echo_client.c

```diff
@@ -0,0 +1,234 @@
+    /* Source address */
+    mp_frame[SRC_ADDR_OFFSET_SHORT_DST] = CLIENT_ADDRESS_LOW_BYTE;
+    mp_frame[SRC_ADDR_OFFSET_SHORT_DST + 1U] = CLIENT_ADDRESS_HIGH_BYTE;
+    /* Data */
+    memcpy(&mp_frame[SRC_ADDR_OFFSET_SHORT_DST + SHORT_ADDRESS_SIZE], m_message, sizeof(m_message));
+}
+
+/** @brief Sets sequence number of a frame in the frame buffer.
+ *
+ *  @param[in] squence_number Sequence number to be set.
+ */
+static void frame_sequence_number_set(uint8_t sequence_number)
+{
+    mp_frame[DSN_OFFSET] = sequence_number;
+}
+
+void nrf_802154_sl_fault_handler(uint32_t module_id, int32_t line, const char * p_error)
+{
+    (void)module_id;
+    (void)line;
+    (void)p_error;
+
+    assert(false);
+}
+
+void nrf_802154_transmitted_raw(const uint8_t * p_frame,
+                                uint8_t       * p_ack,
+                                int8_t          power,
+                                uint8_t         lqi)
+{
+    (void)p_frame;
+    (void)power;
+    (void)lqi;
+
+    nrf_gpio_pin_toggle(LED1);
+
+    if (p_ack != NULL)
+    {
+        nrf_802154_buffer_free_immediately_raw(p_ack);
+    }
+
+    uint32_t current_time = nrf_802154_time_get();
+
+    frame_sequence_number_set(sequence_number_generate());
+    nrf_802154_transmit_raw_at(mp_frame,
+                               true, 
+                               current_time,
+                               NEXT_TX_DELAY_TIME_US,
+                               nrf_802154_channel_get());
+
+    nrf_802154_receive_at(current_time,
+                          RX_DELAY_TIME_US - (WINDOW_WIDTH_US / 2U),
+                          WINDOW_WIDTH_US,
+                          nrf_802154_channel_get());
+
+    nrf_802154_sleep();
+}
+
+void nrf_802154_received_raw(uint8_t * p_data, int8_t power, uint8_t lqi)
+{
+    (void)power;
+    (void)lqi;
+
+    nrf_gpio_pin_toggle(LED2);
+
+    nrf_802154_buffer_free_immediately_raw(p_data);
+    nrf_802154_sleep();
+}
+
+void nrf_802154_transmit_failed(const uint8_t * p_frame, nrf_802154_tx_error_t error)
+{
+    (void)p_frame;
+    (void)error;
+
+    nrf_802154_transmit_raw_at(mp_frame,
+                               true, 
+                               nrf_802154_time_get(),
+                               NEXT_TX_DELAY_TIME_US,
+                               nrf_802154_channel_get());
+
+    nrf_802154_sleep();
+}
+
+int main(void)
+{
+    static uint8_t pan_id[] = {PAN_ID_LOW_BYTE, PAN_ID_HIGH_BYTE};
+    static uint8_t short_address[] = {CLIENT_ADDRESS_LOW_BYTE, CLIENT_ADDRESS_HIGH_BYTE};
+
+    led_init();
+
+    mpsl_utils_init();
+
+    nrf_802154_init();
+    nrf_802154_pan_id_set(pan_id);
+    nrf_802154_channel_set(CHANNEL_802154);
+    nrf_802154_short_address_set(short_address);
+
+    frame_init();
+
+    frame_sequence_number_set(sequence_number_generate());
+    nrf_802154_transmit_raw(mp_frame, true);
+
+    while (1)
+    {
+        __WFE();
+    }
+
+    return 0;
```

Comment:
Unreachable code. Applies to source files of the remaining examples as well

Reply:
Return statement was removed.